deep_quoridor/src/arena.py

```diff
@@ -148,53 +189,59 @@ def _play_game(self, agent1: Agent, agent2: Agent, game_id: str) -> GameResult:
         self.game.close()
         return result
 
-    # Replace the existing _play_games method
-    def _play_games(self, players: list[str | Agent], times: int, mode: PlayMode) -> list[GameResult]:
-        agents = []
-        for p in players:
-            if isinstance(p, Agent):
-                agents.append(p)
-            else:
-                agents.append(AgentRegistry.create_from_encoded_name(p, self.game))
+    def _list_of_games_to_play(self, players: list[str], times: int, mode: PlayMode) -> list[tuple[str, str, str]]:
```

Comment:
I really like this approach of building an explicit list of all the games that need to be played. It simplifies play_games quite a bit, and we could add unit tests for it as some point if we wanted.